src/libexpr/parser.y

```diff
@@ -369,7 +369,7 @@ expr_function
   : ID ':' expr_function
     { $$ = new ExprLambda(CUR_POS, data->symbols.create($1), 0, $3); }
   | '{' formals '}' ':' expr_function
-    { $$ = new ExprLambda(CUR_POS, data->symbols.create(""), toFormals(*data, $2), $5); }
+    { $$ = new ExprLambda(CUR_POS, {}, toFormals(*data, $2), $5); }
```

Comment:
This could use a comment for future readers. A dummy function with a proper name would help a lot.

Reply:
how about
```suggestion
    { $$ = new ExprLambda(CUR_POS, Symbol::none, toFormals(*data, $2), $5); }
```
which would be a new constant akin to `noPos`?

Reply:
added another constructor to ExprLambda instead because C++ really doesn't like constexpr data members of a type to be declared inside the type, and adding a `noSymbol` to the surrounding namespace for a single use seemed rather cluttering.